src/shared/tpm2-util.c

```diff
@@ -2755,6 +2755,7 @@ int tpm2_make_luks2_json(
                                        JSON_BUILD_PAIR_CONDITION(pubkey_pcr_mask != 0, "tpm2_pubkey_pcrs", JSON_BUILD_VARIANT(pkmj)),
                                        JSON_BUILD_PAIR_CONDITION(pubkey_pcr_mask != 0, "tpm2_pubkey", JSON_BUILD_BASE64(pubkey, pubkey_size)),
                                        JSON_BUILD_PAIR_CONDITION(salt, "tpm2_salt", JSON_BUILD_BASE64(salt, salt_size)),
+                                       JSON_BUILD_PAIR_CONDITION(salt, "tpm2_append_salt", JSON_BUILD_BOOLEAN(flags & TPM2_FLAGS_APPEND_SALT)),
```

Comment:
use a more descriptive name, eg `tpm2_append_salted_pin_to_key`

Reply:
Is it okay to clutter up the json-metadata with this?
Another idea I had was to encode it into the salt field, e.g. by prefixing it with a non-base64 char.

Reply:
it exists to carry metadata